core/types/crosslink.go

```diff
@@ -4,51 +4,76 @@ import (
 	"math/big"
 	"sort"
 
-	"github.com/ethereum/go-ethereum/rlp"
-
 	"github.com/ethereum/go-ethereum/common"
-
+	"github.com/ethereum/go-ethereum/rlp"
 	"github.com/harmony-one/harmony/block"
+	"github.com/harmony-one/harmony/shard"
 )
 
 // CrossLink is only used on beacon chain to store the hash links from other shards
+// signature and bitmap correspond to |blockNumber|parentHash| byte array
+// Captial to enable rlp encoding
+// Here we replace header to signatures only, the basic assumption is the committee will not be
+// corrupted during one epoch, which is the same as consensus assumption
 type CrossLink struct {
-	ChainHeader *block.Header
+	ParentHashF  common.Hash
```

Comment:
Let's remove all the "Parent" prefix. Everything here is about the same block. So no need to say parent or not.